Mathlib/GroupTheory/Subsemigroup/Centralizer.lean

```diff
@@ -0,0 +1,183 @@
+/-
+Copyright (c) 2021 Thomas Browning. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Thomas Browning, Jireh Loreaux
+
+! This file was ported from Lean 3 source module group_theory.subsemigroup.centralizer
+! leanprover-community/mathlib commit ffc3730d545623aedf5d5bd46a3153cbf41f6c2c
+! Please do not edit these lines, except to modify the commit id
+! if you have ported upstream changes.
+-/
+import Mathlib.GroupTheory.Subsemigroup.Center
+import Mathlib.Algebra.GroupWithZero.Units.Lemmas
+
+/-!
+# Centralizers of magmas and semigroups
+
+## Main definitions
+
+* `set.centralizer`: the centralizer of a subset of a magma
+* `subsemigroup.centralizer`: the centralizer of a subset of a semigroup
+* `set.add_centralizer`: the centralizer of a subset of an additive magma
+* `add_subsemigroup.centralizer`: the centralizer of a subset of an additive semigroup
+
+We provide `monoid.centralizer`, `add_monoid.centralizer`, `subgroup.centralizer`, and
+`add_subgroup.centralizer` in other files.
+-/
+
+
+variable {M : Type _} {S T : Set M}
+
+namespace Set
+
+variable (S)
+
+/-- The centralizer of a subset of a magma. -/
+@[to_additive add_centralizer " The centralizer of a subset of an additive magma. "]
+def centralizer [Mul M] : Set M :=
+  { c | ∀ m ∈ S, m * c = c * m }
+#align set.centralizer Set.centralizer
```

Comment:
```suggestion
  { c | ∀ m ∈ S, m * c = c * m }
#align set.centralizer Set.centralizer
#align set.add_centralizer Set.addCentralizer
```
`to_additive` doesn't add `#align` statements automatically, so these have to be added manually throughout the file as well.

Reply:
How am I supposed to do the alignment for things such as:
```
@[to_additive "The centralizer of a subset of an additive semigroup."]
def centralizer : Subsemigroup M where
  carrier := S.centralizer
  mul_mem' := Set.mul_mem_centralizer
#align subsemigroup.centralizer Subsemigroup.centralizer
```
There is no explicit naming done here, what is the course of action in this case?

Reply:
@dupuisf ?

Reply:
Guessing the generated name, checking that it is indeed generated, and using that in the `#align` should work.